### PR TITLE
Eugen kehr

### PR DESCRIPTION
**Pull request by 00noize00**
- Fixed .gitignore, will now ignore the node_modules and the build folder. Also changed the include path in the tsconfig.json.
- Finished the bank-account class and extended it with validation functions for input parameters
- Introduced a general validation result interface
- Finished tests in respect to the changes in the bank-account class with 100% coverage
- Removed node-modules from the repository

### CODE REVIEW

File: coding-challenge/tst/bank/bank-account.test.ts [hunk 1]
```diff
@@ -30,41 +30,72 @@ describe('Tests for bank account class', () => {
...
         // Arrange
         const withdrawAmount = -5;
+        const error = new Error('Withdraw amount has to be greater than 0!');
 
         // Act & Assert
-        expect(() => {
-            bankAccount.withdraw(withdrawAmount)
-        }).toThrow('Withdraw amount has to be greater than 0!');
+        try {
+            bankAccount.withdraw(withdrawAmount);
```
Comment (Chrisissorry) [inline coding-challenge/tst/bank/bank-account.test.ts:37]: Why is it better to use a try/catch here?

Comment (00noize00) [reply coding-challenge/tst/bank/bank-account.test.ts]: With try/catch we can disect the error and can write with that a more precise test if needed.

Nevertheless I just found a problem in the code, fixed it and also extended the test cases. With that I changed the try/catch approach to the .toThrow() method, in order to keep it more simple and maybe even more readable.

File: coding-challenge/src/bank/bank-account.ts [hunk 2]
```diff
@@ -18,6 +18,8 @@ export class BankAccount {
         
         if (validationResult.isValid) {
             this.balance -= withdrawAmount;
+        } else {
```
Comment (Chrisissorry) [inline coding-challenge/src/bank/bank-account.ts:21]: @00noize00 Just FYI: We have a rule here to never use `else` in your code ;-)

Comment (00noize00) [reply coding-challenge/src/bank/bank-account.ts]: Okay, I can handle that easily :)

**Comment by adrianmota-sag** [comment]
The private methods created to do the actual validation are a bit of over engineering, you had an example of the validation on one of the class methods. Maybe if a single method to do the validation in both places would’ve been used it would be a different story.
Some of the existing tests were removed which was not part of the exercise.
Inconsistencies regarding constants naming, it() or test() functions in jest.
We should not test private methods in order not to break the encapsulation principle.

**Comment by Chrisissorry** [comment]
@00noize00 Thanks for your submission. Above we provided some feedback. Closing this PR as done :-)

**Pull request closed without merging**